models/tsodyks_connection_hom.h

```diff
@@ -0,0 +1,331 @@
+  ~TsodyksConnectionHom()
+  {
+  }
+
+  // Explicitly declare all methods inherited from the dependent base ConnectionBase.
+  // This avoids explicit name prefixes in all places these functions are used.
+  // Since ConnectionBase depends on the template parameter, they are not automatically
+  // found in the base class.
+  using ConnectionBase::get_delay_steps;
+  using ConnectionBase::get_rport;
+  using ConnectionBase::get_target;
+
+  /**
+   * Get all properties of this connection and put them into a dictionary.
+   */
+  void get_status( DictionaryDatum& d ) const;
+
+  /**
+   * Set properties of this connection from the values given in dictionary.
+   */
+  void set_status( const DictionaryDatum& d, ConnectorModel& cm );
+
+  /**
+   * Send an event to the receiver of this connection.
+   * \param e The event to send
+   * \param t_lastspike Point in time of last spike sent.
+   * \param cp Common properties to all synapses (empty).
+   */
+  void send( Event& e, thread t, double_t t_lastspike, const TsodyksHomCommonProperties& cp );
+
+  class ConnTestDummyNode : public ConnTestDummyNodeBase
+  {
+  public:
+    // Ensure proper overriding of overloaded virtual functions.
+    // Return values from functions are ignored.
+    using ConnTestDummyNodeBase::handles_test_event;
+    port
+    handles_test_event( SpikeEvent&, rport )
+    {
+      return invalid_port_;
+    }
+  };
+
+  void
+  check_connection( Node& s, Node& t, rport receptor_type, double_t, const CommonPropertiesType& )
+  {
+    ConnTestDummyNode dummy_target;
+    ConnectionBase::check_connection_( dummy_target, s, t, receptor_type );
+  }
+
+  void set_weight( double_t )
+  {
+    throw BadProperty(
+      "Setting of individual weights is not possible! The common weights can be changed via "
+      "CopyModel()." );
+  }
+
+private:
+  double_t x_; //!< amount of resources in recovered state
+  double_t y_; //!< amount of resources in active state
+  double_t u_; //!< actual probability of release
+};
+
+
+/**
+ * Send an event to the receiver of this connection.
+ * \param e The event to send
+ * \param p The port under which this connection is stored in the Connector.
+ * \param t_lastspike Time point of last spike emitted
+ */
+template < typename targetidentifierT >
+inline void
+TsodyksConnectionHom< targetidentifierT >::send( Event& e,
+  thread t,
+  double_t t_lastspike,
+  const TsodyksHomCommonProperties& cp )
+{
+  double_t h = e.get_stamp().get_ms() - t_lastspike;
+
+  // t_lastspike_ = 0 initially
+  // this has no influence on the dynamics, IF y = z = 0 initially
+  // !!! x != 1.0 -> z != 0.0 -> t_lastspike_=0 has influence on dynamics
+
+  // propagator
+  // TODO: use expm1 here instead, where applicable
+  double_t Puu = ( cp.tau_fac_ == 0.0 ) ? 0.0 : std::exp( -h / cp.tau_fac_ );
+  double_t Pyy = std::exp( -h / cp.tau_psc_ );
+  double_t Pzz = std::exp( -h / cp.tau_rec_ );
+
+  // double_t Pzy = (Pyy - Pzz) * tau_rec_ / (tau_psc_ - tau_rec_);
```

Comment:
Could you remove this dead code? It should also be removed from the original tsodyks-synapse code.